### PR TITLE
dashless updates

### PR DESCRIPTION
**Pull request by nebs13**
1A:
6z ReqBoth changed to ReqOut - dash only required to enter "left" subroom
8 ReqIn 0 dash hard
8 ReqOut changed to ReqBoth (Down Idx:1) - impossible to get above crumble blocks dashless
7a add ReqOut (Up Idx:0) for 1 dash (hard) - can be done with an updash

1B:
01 reverse 0 dash nerfed to normal
02 add 0 dash to ReqOut (perfect) - performed RTA with slowdown
04 reverse add 0 dash (expert) - has a couple spike jumps but not too bad
07 add 0 dash to ReqOut (expert)
end removed 0 dash ReqIn - will always collide with spikes after collecting heart, could be fixed with a tweak?

1C:
02 add 0 dash to ReqIn (Down Idx:5 and Down Idx:6 - perfect) - Idx:5 likely never appears, Idx:6 experiences softlocking issues when entering directly under the zipper, required a double cornerboost to collect the heart without dying

Potentially a couple other things I forgot to write down

Suggestions:
Quick map regeneration option from pause menu - could take the same settings but use a new seed to generate a map
1A 8 tweak crumble block by Left Idx:0, change to Kind: inout
1A 10 tweak crumble blocks by Up Idx:1, change to Kind: inout
1B end tweak heart position to make dashless possible
1C 02 tweak coin block position - would make whole room viable for rta dashless

### CODE REVIEW

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml [hunk 1]
```diff
@@ -533,7 +526,10 @@ ASide:
     Holes:
     - Side: Up
       Idx: 0
-      Kind: in
+      Kind: inout
+      ReqOut:
+        Dashes: one
+        Difficulty: hard
```
Comment (rhelmot) [inline Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:532]: I specifically don't want to make this change; finding the perfect pixel to dash upwards with no corner correction sounds extremely not fun.

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml [hunk 2]
```diff
@@ -487,22 +485,17 @@ ASide:
...
+      ReqBoth:
         Or:
         - Dashes: one
         - Dashes: zero
           Difficulty: expert
-      ReqIn:
-        Or:
-        - Dashes: one
-        - Dashes: zero
-          Difficulty: hard
```
Comment (rhelmot) [inline Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:505]: Why do you think this shouldn't be in hard? it's just a coyote jump off the crumble block to get under the wall; I pulled it off twice in my first three attempts just now

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml [hunk 3]
```diff
@@ -221,8 +221,6 @@ ASide:
     - To: "left"
       ReqOut:
         Dashes: one
-      ReqIn:
-        Dashes: zero
```
Comment (rhelmot) [inline Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:225]: no, uh, this is definitely doable dashless. you just have to jump off the top block and fall to the right side. It's super easy.

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml [hunk 4]
```diff
@@ -620,12 +616,6 @@ ASide:
       - Side: Right
         Idx: 0
         Kind: inout
-      InternalEdges:
-      - Collectable: 0
-        ReqOut:
-          Dashes: one
-        ReqIn:
-          Dashes: zero
```
Comment (rhelmot) [inline Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:628]: why is this removed? Did you not rebase your changes on the latest master?

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml [hunk 5]
```diff
@@ -812,30 +802,30 @@ ASide:
...
-    - Room: "top"
-      Holes:
-      - Side: Left
-        Idx: 0
-        Kind: inout
-    - Room: "right"
-      Holes:
-      - Side: Right
-        Idx: 0
-        Kind: inout
```
Comment (rhelmot) [inline Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:838]: yeah this is a mess. I think you did your changes on an older version and then just copied your old file into the new branch instead of rebasing properly. if I were to merge this rn it would revert a few days of tweaks.

Comment (nebs13) [reply Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:838]: I based the changes on the bundled config with 0.4.8 - will redo based on git version

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml (hunk 3 above)
Comment (nebs13) [reply Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:225]: thought these lines were redundant since its basically equivalent to saying theres no requirement. may be misunderstanding if both ReqOut and ReqIn are required

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml (hunk 2 above)
Comment (nebs13) [reply Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:505]: modified the wrong one whoops

File: Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml (hunk 1 above)
Comment (nebs13) [reply Randomizer/Config/Celeste/1-ForsakenCity.rando.yaml:532]: valid

**Review by nebs13 — COMMENTED** [review]
rebased changes on current git version:
- most changes to 1b had already been made
- 1a 8zb reverse 0 dash buffed to expert (i seriously struggled with this the neutrals at the end feel too hard for hard)
- 1a 8 ReqOut changed to ReqBoth for Down Idx: 1 - can't get on top of the crumble block dashless
- 1b 01 0 dash reverse nerfed to hard - got it repeatedly
- 1c 02 Down Idx:5 and Down Idx:6 changed Kind to unknown, added ReqIn
  - Idx:5 may not work properly just because the hole is so wide but figured it was worth trying
  - heart is collectable 0 dash with a double cornerboost, so added to perfect difficulty

**Comment by rhelmot** [comment]
Looks great. I agree entirely with your difficulty rankings.
